cura/Settings/ContainerManager.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2016 Ultimaker B.V.
+# Cura is released under the terms of the AGPLv3 or higher.
+
+from PyQt5.QtCore import QObject, pyqtSlot, pyqtProperty, pyqtSignal
+
+import UM.Settings
+
+from UM.i18n import i18nCatalog
+catalog = i18nCatalog("cura")
+
+class ContainerManager(QObject):
+    def __init__(self, parent = None):
+        super().__init__(parent)
+
+        self._registry = UM.Settings.ContainerRegistry.getInstance()
+
+    @pyqtSlot(str)
+    def duplicateContainer(self, container_id):
+        containers = self._registry.findInstanceContainers(id = container_id)
+        if not containers:
+            return
+
+        new_name = self_registry.uniqueName(containers[0].getName())
+        new_material = containers[0].duplicate(new_name)
+        self._registry.addContainer(new_material)
+
+    @pyqtSlot(str, str)
+    def renameContainer(self, container_id, new_name):
+        pass
+
+    @pyqtSlot(str)
+    def removeContainer(self, container_id):
+        pass
+
+    @pyqtSlot(str, str, str)
+    def setContainerMetaDataEntry(self, container_id, entry_name, entry_value):
```

Comment:
In other places we've held the convention (originating from @awhiemstra too) that the Data in metadata is not capitalised; i.e. it was not considered a new word.